things/service/src/main/java/org/eclipse/ditto/things/service/enforcement/RollbackCreatedPolicy.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.ditto.things.service.enforcement;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.ditto.base.model.common.ResponseType;
+import org.eclipse.ditto.base.model.signals.Signal;
+import org.eclipse.ditto.things.model.signals.commands.modify.CreateThing;
+import org.eclipse.ditto.things.model.signals.commands.modify.CreateThingResponse;
+
+
+public record RollbackCreatedPolicy(Signal<?> initialCommand, Object response, CompletableFuture<Object> responseFuture) {
+
+    public static RollbackCreatedPolicy of(final Signal<?> command, final Object response,
+            final CompletableFuture<Object> responseFuture) {
+        return new RollbackCreatedPolicy(command, response, responseFuture);
+    }
+
+    public static boolean shouldRollback(final Signal<?> command, final Object response) {
+        return  command instanceof CreateThing && response instanceof CreateThingResponse thingResponse &&
+                thingResponse.getResponseType() == ResponseType.ERROR;
```

Comment:
I don't think that this will work.
`CreateThingResponse` is only created for the "success case".
In case of an error, you won't get a `CreateThingResponse`,but probably a `DittoRuntimeException`, more specifically a `ThingException`.
Or you would get a `ThingErrorResponse` (I currently don't remember where a DittoRuntimeException is converted to a `ThingErrorResponse`, but I think at a later point).

I wonder if you tested this functionality and whether it did work with this current implemetation?

Reply:
I was not sure here either that is why a wanted to hear your opinion on that. Maybe i should have left a comment about that.